main.py

```diff
@@ -326,37 +345,38 @@ def quantize_aq(
 
         # Logging
         stats_payload["layer_time"] = time.time() - start_time
-        stats_payload["out_loss"] = torch.mean(torch.Tensor(out_losses)).item()
-        if run_validation:
-            stats_payload["out_val_loss"] = torch.mean(torch.Tensor(out_val_losses)).item()
         stats_payload["Step"] = layer_index
         if args.wandb:
             wandb.log(stats_payload, step=layer_index)
-        print(stats_payload)
+        if not loaded_layer:
+            print(stats_payload)
 
     print("=====================\nFinal stats:")
     if args.save:
         torch.save(vars(args), os.path.join(args.save, "args.pt"))
         save_not_quantized_weights(model, args.save)
+        if args.on_save:
+            exec(args.on_save)  # a callback e.g. to save progress in slurm or similar distributed infrastructure
 
     if args.wandb:
         wandb.log({"max_cuda_mem_quantize": round(torch.cuda.max_memory_allocated() / 1e9, 2)})
-        wandb.log({"Avg_bits": overall_bits / model_number_of_params})
+        if number_of_quantized_params > 0:  # do not report avg bits if we load all pre-quantized layers via --resume
+            wandb.log({"Avg_bits": overall_bits / number_of_quantized_params})
     model.config.use_cache = use_cache
     print(f"quantize: {torch.cuda.max_memory_allocated()=:,}")
     return quantizers
 
 
 @torch.no_grad()
-def perplexity_eval(model, testenc, args):
+def perplexity_eval(model: PreTrainedModel, testenc: torch.LongTensor, args: Namespace) -> float:
```

Comment:
I tested that running perplexity_eval with old vs new get_inps returns exactly the same perplexity